code/__HELPERS/game.dm

```diff
@@ -12,6 +12,36 @@
 	return format_text ? format_text(A.name) : A.name
 
 
+/// Checks all conditions if a spot is valid for construction , will return TRUE
+/proc/IsValidForResinStructure(turf/target, needsSupport = FALSE, mob/builder)
```

Comment:
DM doesnt have a strong type system, but it is no reason to do what you are doing.

You cannot return either a string or an int
Make all your return value be an int